### PR TITLE
Move AbortOnResupply from AttackAircraft to AttackFollow

### PR DESCRIPTION
**Pull request by PunkPun**
I give the ability for non-AttackAircraft actors to resupply on attack

 I basically just copied over the code from AttackAircraft to AttackFollow

### CODE REVIEW

**Comment by PunkPun** [comment]
Partially solves #18645

**Comment by PunkPun** [comment]
Testcase
[ascentt.oramap.zip](https://github.com/OpenRA/OpenRA/files/9149644/ascentt.oramap.zip)

I add ammo to RA medium tank and make Longbow use AttackTurreted

File: OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs [hunk 1]
```diff
@@ -343,6 +356,39 @@ public override bool Tick(Actor self)
...
+
+					// AbortOnResupply cancels the current activity (after resupplying) plus any queued activities
+					if (attack.Info.AbortOnResupply)
+						NextActivity?.Cancel(self);
+
+					if (isAircraft)
+						QueueChild(new ReturnToBase(self));
+					else
+					{
+						if (rearmable != null && attack.Info.AbortOnResupply)
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs]: `rearmable != null` is always true here.

Comment (PunkPun) [reply OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs]: fixed

File: OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs [hunk 2]
```diff
@@ -343,6 +356,39 @@ public override bool Tick(Actor self)
...
+
+					// AbortOnResupply cancels the current activity (after resupplying) plus any queued activities
+					if (attack.Info.AbortOnResupply)
+						NextActivity?.Cancel(self);
+
+					if (isAircraft)
+						QueueChild(new ReturnToBase(self));
+					else
+					{
+						if (attack.Info.AbortOnResupply)
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs]: Why is it doing this just if we are aborting? It should do the same as for aircraft: Resupply and act according to `AbortOnResupply` only after that.

**Review by abcdefg30 — COMMENTED** [review]
Lgtm otherwise.

File: OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs (hunk 2 above)
Comment (PunkPun) [reply OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs]: That does make sense

**Review by abcdefg30 — DISMISSED** [review]

File: OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs [hunk 3]
```diff
@@ -239,16 +246,20 @@ class AttackActivity : Activity, IActivityNotifyStanceChanged
...
 				attack = self.Trait<AttackFollow>();
 				move = allowMove ? self.TraitOrDefault<IMove>() : null;
 				revealsShroud = self.TraitsImplementing<RevealsShroud>().ToArray();
+				rearmable = self.TraitOrDefault<Rearmable>();
 
 				this.target = target;
 				this.forceAttack = forceAttack;
 				this.targetLineColor = targetLineColor;
+				this.source = source;
+				isAircraft = self.TraitOrDefault<Aircraft>() != null;
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs]:
```suggestion
				isAircraft = self.Info.HasTraitInfo<AircraftInfo>();
```

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/c79fc42930a6920a61b6b7e3213e2e5aa60afc78...96d92645f486670797a4845cfc2d2e42a3b438ce)

**Pull request merged**